### ISSUE TITLE
Different results in stable diffusion and diffusers

### ISSUE THREAD

**Issue by Laidawang** [huggingface/diffusers#4460]
I can get the same result in stable diffusion and diffusers using txt2img.
diffusers
![image](https://github.com/huggingface/diffusers/assets/85244566/3e7ea8ef-e10e-4e74-a3c5-79bd5d6d7cb7)
stable diffusion webui
![image](https://github.com/huggingface/diffusers/assets/85244566/444c1d37-2e06-4276-8409-5a912c4be827)

However, the situation becomes different when I add controlnet.
diffusers
![image](https://github.com/huggingface/diffusers/assets/85244566/2f12b0ae-b421-4a26-879c-15db271cae62)
webui
![image](https://github.com/huggingface/diffusers/assets/85244566/b7dd79dc-b6a1-491d-8995-95f8975244fd)

here is my code in diffusers
```
from diffusers import (
    StableDiffusionPipeline,
    StableDiffusionControlNetPipeline, 
    ControlNetModel, 
    UniPCMultistepScheduler, 
    DPMSolverMultistepScheduler, 
    EulerDiscreteScheduler
)
from diffusers.utils import load_image
import numpy as np
import torch

import cv2
from PIL import Image

image = load_image(
    "/data2/home/srchen/project/diffusers_all/compare_diffusers_webui/txt2img_controlnet/test.png"
)
image = np.array(image)
image = cv2.Canny(image, 100, 200)
image = image[:, :, None]
image = np.concatenate([image, image, image], axis=2)
canny_image = Image.fromarray(image)
canny_image = load_image("/data2/home/srchen/project/diffusers_all/compare_diffusers_webui/txt2img_controlnet/canny.png")



base_pipe = StableDiffusionPipeline.from_single_file(
    "/data2/home/srchen/sd-models/realisticVisionV30_v30VAE.safetensors",
    torch_dtype=torch.float16,
    safety_checker=None,
).to("cuda")

controlnet = ControlNetModel.from_pretrained(
                                            "/data2/home/srchen/project/diffusers_all/compare_diffusers_webui/txt2img_controlnet/sd-controlnet-canny",
                                            torch_dtype=torch.float16,
                                            ).to("cuda")

pipe = StableDiffusionControlNetPipeline(
            vae=base_pipe.vae,
            text_encoder=base_pipe.text_encoder,
            tokenizer=base_pipe.tokenizer,
            unet=base_pipe.unet,
            controlnet=controlnet,
            scheduler=base_pipe.scheduler,
            safety_checker=None,
            feature_extractor = base_pipe.feature_extractor,
            requires_safety_checker = False,
            )

pipe.scheduler = EulerDiscreteScheduler.from_config(pipe.scheduler.config)

control_weight = 0.1

for i in range(10):
    control_weight += 0.1
    images = pipe(
        image = canny_image,
        prompt="a beautiful shoe",
        negative_prompt="", 
        width=512,
        height=512,
        num_inference_steps=20,
        num_images_per_prompt=1,
        guidance_scale=7,
        controlnet_conditioning_scale = control_weight,
        generator=torch.Generator(device='cuda').manual_seed(10)
    ).images
    images[0].save(f"diffusers_compare_txt2img_{control_weight}.png")
```

and settings in webui
![image](https://github.com/huggingface/diffusers/assets/85244566/ae303843-bcd7-49b7-82dc-9ab08c90dbce)

I wonder what makes the difference.Is it a different implementation, or is there some special controlnet setting I'm not enabling？

**Comment by slep0v** [comment]
Having the same issue especially with controlnet + img2img pipeline
Diffusers
![image](https://github.com/huggingface/diffusers/assets/37597789/89532b93-cf13-4131-be35-ce22e90eb120)

Automatic1111 webui
![image](https://github.com/huggingface/diffusers/assets/37597789/30fe51df-98b4-4850-8253-58ee6eacb89f)

**Comment by Laidawang** [comment]
I found some differences, I hope it can help you solve it.
1. The sampler uses euler, 
2. sdwebui has re-weighted the prompt.
3. The generate in diffusers does not represent the same value as the webui seed. In img2img (diffusers), generate will be called once in vae, so you should reset generate after that to ensure that the seed for creating random noise is consistent.
